### ISSUE TITLE
this widget has been unmounted, so the State no longer has a context (and should be considered defunct)

### ISSUE THREAD

**Issue by suhaib-raadan** [koukibadr/Cherry-Toast#34]
why this happen sometimes in my app

**Comment by koukibadr** [comment]
Hello @suhaib-raadan can you share more details ? (code example, screenshots...)

**Comment by alihassan143** [comment]
@suhaib-raadan  you are calling toast when the actual context of the widget is no longer mounted  you can use condition like
```
if(context.mounted){
// show toast 
}
```

**Comment by crimson-gao** [comment]
Use a dialog to trigger a toast (using homePage context) and go back to homePage. While the toast is showing, if we open another dialog, the toast would be defunct and cause exceptions as above.

**Comment by shuzibanshou** [comment]
> @suhaib-raadan you are calling toast when the actual context of the widget is no longer mounted you can use condition like
> 
> ```
> if(context.mounted){
> // show toast 
> }
> ```

Yeah, I found this issue as well, when the toast widget auto dismiss if you click the widget it will report this error, but it don't appear everytime

should modify the 333 line of 《cherry_toast.dart》
`          if(mounted){
            Navigator.maybePop(context);
          }`

**Comment by koukibadr** [comment]
Checkout the new version 1.2.1 contains the bug fix
[https://pub.dev/packages/cherry_toast](https://pub.dev/packages/cherry_toast)
